### PR TITLE
Adding notebook showing how to use tool_required

### PR DESCRIPTION
**Pull request by colin-openai**
## Summary

Adds a notebook demonstrating how to use tool_required to make a basic customer service assistant.

## Motivation

We have just released the tool_required parameter and it is a great unlock for users who want a more deterministic Assistant experience that is channelled with tools.

---

## For new content

When contributing new content, read through our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md), and mark the following action items as completed:

- [X] I have added a new entry in [registry.yaml](https://github.com/openai/openai-cookbook/blob/main/registry.yaml) (and, optionally, in [authors.yaml](https://github.com/openai/openai-cookbook/blob/main/authors.yaml)) so that my content renders on the cookbook website.
- [X] I have conducted a self-review of my content based on the [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md#rubric):
  - [X] Relevance: This content is related to building with OpenAI technologies and is useful to others.
  - [X] Uniqueness: I have searched for related examples in the OpenAI Cookbook, and verified that my content offers new insights or unique information compared to existing documentation.
  - [X] Spelling and Grammar: I have checked for spelling or grammatical mistakes.
  - [X] Clarity: I have done a final read-through and verified that my submission is well-organized and easy to understand.
  - [X] Correctness: The information I include is correct and all of my code executes successfully.
  - [X] Completeness: I have explained everything fully, including all necessary references and citations.

We will rate each of these areas on a scale from 1 to 4, and will only accept contributions that score 3 or higher on all areas. Refer to our [contribution guidelines](https://github.com/openai/openai-cookbook/blob/main/CONTRIBUTING.md) for more details.

### CODE REVIEW

File: examples/Using_tool_required_for_customer_service.ipynb [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+    "The `ChatCompletion` endpoint now includes the ability to specify whether a tool **must** be called every time. This adds an element of determinism to how you build your wrapping application, as you can count on a tool being provided with every call. \n",
+    "\n",
+    "We'll demonstrate here how this can be useful for a contained flow like customer service, where having the ability to define specific exit points gives more control.\n",
+    "\n",
+    "The notebook concludes with a multi-turn evaluation, where we spin up a customer GPT to imitate our customer and test the LLM customer service agent we've set up."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
```
Comment (teomusatoiu) [inline examples/Using_tool_required_for_customer_service.ipynb:19]: can you run the cells all from the start to have them display as 1, 2, 3 ....?

File: examples/Using_tool_required_for_customer_service.ipynb [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+     "text": [
+      "User: Hi, I have had an item stolen that was supposed to be delivered to me yesterday.\n",
+      "Calling function get_instructions\n",
+      "Calling function speak_to_user\n",
+      "Assistant: I'm sorry to hear about the stolen item. Could you please provide me with more details about the fraudulent activity, including the date and the items involved? This information will help us to investigate further and process any necessary actions, such as a refund.\n"
+     ]
+    }
+   ],
+   "source": [
+    "messages = submit_user_message(\"Hi, I have had an item stolen that was supposed to be delivered to me yesterday.\",messages)"
```
Comment (teomusatoiu) [inline examples/Using_tool_required_for_customer_service.ipynb:250]: messages is not defined

**Review by teomusatoiu — CHANGES_REQUESTED** [review]
Everything else looks good! Thanks for pushing this, i think it'll be very useful!

File: examples/Using_tool_required_for_customer_service.ipynb (hunk 1 above)
Comment (colin-openai) [reply examples/Using_tool_required_for_customer_service.ipynb]: This is now done

File: examples/Using_tool_required_for_customer_service.ipynb (hunk 2 above)
Comment (colin-openai) [reply examples/Using_tool_required_for_customer_service.ipynb]: Good catch, updated - not defining it in first cell so the function uses the default.

**Review by teomusatoiu — APPROVED** [review]
All looks good!

File: examples/Using_tool_required_for_customer_service.ipynb
Comment (sirw20035-hash) [inline examples/Using_tool_required_for_customer_service.ipynb]: custom-gpt
,
actions
,
android-app

**Comment by sirw20035-hash** [comment]
@cookbook

**Pull request merged**